source/as-promise.ts

```diff
@@ -7,17 +7,17 @@ import {normalizeArguments, mergeOptions} from './normalize-arguments';
 import requestAsEventEmitter, {proxyEvents} from './request-as-event-emitter';
 import {CancelableRequest, GeneralError, NormalizedOptions, Response} from './types';
 
-const parseBody = (body: Buffer, responseType: NormalizedOptions['responseType'], encoding: NormalizedOptions['encoding']): unknown => {
+const parseBody = (kludge: {body: Buffer}, responseType: NormalizedOptions['responseType'], encoding: NormalizedOptions['encoding']): unknown => {
```

Comment:
don't do this

Reply:
@szmarczak please elaborate. My intent wasn't to commit a proper fix but to hint at what and how this issue occurs.

I never expected this kludge to pollute the codebase and couldn't decide on a better approach.

Reply:
Perhaps a better resolution would be to:

- call the `shortcut()` factory function with a reference to the `promise` variable
- then from within `shortcut()` to bind the resulting buffer()/json()/text() function to said `promise` reference
- finally when invoked, buffer()/json()/text() function will `await` its `this` in order to access the `response`, next the `body`

I can give it a try if you think that approach is acceptable.